server/sonar-server/src/main/java/org/sonar/server/computation/debt/DebtModelHolder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.debt;
+
+import java.util.Collection;
+import javax.annotation.CheckForNull;
+
+public interface DebtModelHolder {
+
+  /**
+   * Return a characteristic by its key
+   *
+   * @throws IllegalStateException if the holder is empty
+   */
+  @CheckForNull
+  Characteristic characteristicByKey(String key);
```

Comment:
Indeed, it's more explicit to add a prefix. 
And I could also use an Optional, I'm still not used to use it.